snap/snapcraft.yaml.in

```diff
@@ -0,0 +1,263 @@
+      export GOPATH=$(realpath ../go)
+      cd ${GOPATH}/src/github.com/${SNAPCRAFT_PROJECT_NAME}/ksm-throttler
+      make TARGET=kata-ksm-throttler
+      make install \
+        DESTDIR=${SNAPCRAFT_PART_INSTALL} \
+        TARGET=kata-ksm-throttler
+
+  kernel:
+    source: @KERNEL_URL@/linux-@KERNEL_VERSION@.tar.xz
+    source-type: tar
+    after: [kernel-dump]
+    plugin: kernel
+    override-build: |
+      case "$(arch)" in
+        "x86_64")
+          config=x86_64_kata_kvm_4.14.x
+        ;;
+
+        "ppc64le")
+          config=ppc64le_kata_kvm_4.14.x
+        ;;
+
+        "aarch64")
+          config=arm64_kata_kvm_4.14.x
+        ;;
+
+        *)
+          echo "ERROR: Unsupported architecture $(arch)"
+          exit 1
+        ;;
+      esac
+
+      make mrproper
+
+      # Apply patches
+      for patch in ${SNAPCRAFT_STAGE}/kernel/patches/*.patch; do
+        echo "Applying $(basename "$patch") ..."
+        patch \
+          --batch \
+          --forward \
+          --strip 1 \
+          --input "$patch"
+        echo
+      done
+
+      # Copy config file
+      cp ${SNAPCRAFT_STAGE}/kernel/configs/${config} .config
+      make -s oldconfig EXTRAVERSION=".container" > /dev/null
+      make -j $(nproc) EXTRAVERSION=".container"
+
+      kernel_suffix=@KERNEL_VERSION@.container
+      kata_kernel_dir=${SNAPCRAFT_PART_INSTALL}/usr/share/kata-containers
+      mkdir -p ${kata_kernel_dir}
+
+      # Install bz kernel
+      make install INSTALL_PATH=${kata_kernel_dir} EXTRAVERSION=".container" || true
+      vmlinuz_name=vmlinuz-${kernel_suffix}
+      ln -sf ${vmlinuz_name} ${kata_kernel_dir}/vmlinuz.container
+
+      # Install raw kernel
+      vmlinux_name=vmlinux-${kernel_suffix}
+      cp vmlinux ${kata_kernel_dir}/${vmlinux_name}
+      ln -sf ${vmlinux_name} ${kata_kernel_dir}/vmlinux.container
+
+  kernel-dump:
+    source: kernel
+    plugin: dump
+    organize:
+      '*' : kernel/
+    prime:
+      - -*
+
+  qemu:
+    source: https://github.com/kata-containers/qemu/archive/@QEMU_LITE_BRANCH@.tar.gz
+    source-type: tar
+    plugin: make
+    after: [qemu-scripts-dump, qemu-patches-dump]
+    build-packages:
+      - gcc
+      - python
+      - zlib1g-dev
+      - libcap-ng-dev
+      - libglib2.0-dev
+      - libpixman-1-dev
+      - libnuma-dev
+      - libltdl-dev
+      - libcap-dev
+      - libattr1-dev
+    override-build: |
+      chmod +x ${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh
+      # static build
+      echo "$(${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh -s qemu) \
+        --disable-rbd
+        --prefix=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr \
+        --datadir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/share \
+        --libexecdir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/libexec/qemu" \
+        | xargs ./configure
+
+      git clone https://github.com/qemu/keycodemapdb ui/keycodemapdb/
+      cd ui/keycodemapdb; git checkout 10739aa; cd ../..
```

Comment:
Do we not hold these magic SHAs in our versions.yaml file - feels like maybe we should? /cc @jodh-intel @jcvenegas

Reply:
Oooh - good catch - I'd missed that ;)